pkg/lifecyclehooks/lyfecyclehooks.go

```diff
@@ -0,0 +1,207 @@
+						},
+					},
+				},
+			},
+			BackoffLimit: pointer.Int32Ptr(4),
+			Completions:  pointer.Int32Ptr(njobs),
+			Parallelism:  pointer.Int32Ptr(njobs),
+		},
+	}
+	if nodeSelector != "" {
+		job.Spec.Template.Spec.NodeSelector = map[string]string{
+			nodeSelector: "",
+		}
+	}
+	return job
+}
+
+var _ = Describe("[Feature:Machine] Lifecycle Hooks should", func() {
+	var client runtimeclient.Client
+	var machineSet *v1beta1.MachineSet
+	var machine *v1beta1.Machine
+	var workload *batchv1.Job
+	var pod *corev1.Pod
+	ctx := context.Background()
+	var err error
+
+	BeforeEach(func() {
+		By("Creating the machineset")
+		client, err = framework.LoadClient()
+		// Build machine set parameters
+		expectedReplicas := 1
+		machineSetParams := framework.BuildMachineSetParams(client, expectedReplicas)
+		// Create a label for node and add to machine set parameters
+		targetedNodeLabel := fmt.Sprintf("%v", LifecyclehooksWorkerNodeRoleLabel)
+		machineSetParams.Labels[targetedNodeLabel] = ""
+		// Create machine set
+		machineSet, err = framework.CreateMachineSet(client, machineSetParams)
+		Expect(err).ToNot(HaveOccurred())
+		// Wait for machine to be running
+		framework.WaitForMachineSet(client, machineSet.GetName())
+		By("Running a workload on the machine")
+		// Run a pod on this machine
+		workloadMemRequest := resource.MustParse("100000000")
+		workload = newWorkLoad(int32(expectedReplicas), workloadMemRequest, targetedNodeLabel)
+		Expect(client.Create(ctx, workload)).Should(Succeed())
+		Expect(err).ToNot(HaveOccurred())
+		By("Waiting for job pod to start running on machine.")
+		Eventually(func() bool {
+			jobPodList, err := framework.GetPods(client, map[string]string{lifecycleHooksTestLabel: ""})
+			Expect(err).ToNot(HaveOccurred())
+			pod = &jobPodList.Items[0]
+			return len(jobPodList.Items) == int(expectedReplicas) && pod.Status.Phase == corev1.PodRunning
+		}, framework.WaitMedium, pollingInterval).Should(BeTrue())
+	})
+
+	AfterEach(func() {
+		By("Deleting the machineset")
+		cascadeDelete := metav1.DeletePropagationForeground
+		Expect(client.Delete(ctx, machineSet, &runtimeclient.DeleteOptions{
+			PropagationPolicy: &cascadeDelete,
+		})).To(Succeed())
+		By("Deleting workload job")
+		Expect(client.Delete(ctx, workload, &runtimeclient.DeleteOptions{
+			PropagationPolicy: &cascadeDelete,
+		}))
+	})
+
+	It("pause lifecycle actions when present", func() {
+		By("Setting lifecycle hooks on the machine")
+		// Create a pre-drain hook for the machine
+		predrainHook := &v1beta1.LifecycleHook{
+			Name:  "cluster-api-actuator-pkg/pre-drainHook",
+			Owner: "cluster-api-actuator-pkg",
+		}
+		machines, err := framework.GetMachinesFromMachineSet(client, machineSet)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(machines).To(HaveLen(1))
+		machine = machines[0]
+		machine.Spec.LifecycleHooks.PreDrain = []v1beta1.LifecycleHook{*predrainHook}
+		err = client.Update(context.Background(), machine)
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
```suggestion
		Expect(client.Update(context.Background(), machine)).To(Succeed())
```
when function only returns an error you can check the error value without assigning it to a variable